test-app/pages/index.tsx

```diff
@@ -18,38 +18,21 @@ let textField = text({
 
     return validation.valid(value);
   },
-  stateFromChange: (changed, current) => {
-    if (changed.value === "something") {
+  stateFromChange: (current, next) => {
```

Comment:
It looks like I messed up stateFromChange a bit, it should be the next state as the first arg and the prev state as the second arg. It looks like object stateFromChange is being called with args in the right order but has the wrong names in the types and scalar stateFromChange is called in the wrong order and has the right names for that wrong order(so the names need to be changed too). Could you fix that?